src/org/jitsi/impl/neomedia/rtp/TransportCCEngine.java

```diff
@@ -361,5 +386,70 @@ private RTCPTransformer()
         }
     }
 
+    /**
+     * {@Link BITRATE_UPDATE_INTERVAL} is an interval used to
+     */
+    private static final int BITRATE_UPDATE_INTERVAL = 5;
+
+    private class PacketDetail {
+        /**
+         * {@Link packetDetail} is an object that holds the
+         * length(size) of the packet in {@Link pktLength}
+         * and the time stamps of the outgoing packet
+         * in {@Link pktTime}
+         */
+        int pktLength;
+        long pktTime;
+        PacketDetail(int length, long time){
+            pktLength = length;
+            pktTime  = time;
+        }
+    }
+
+    /**
+     * Calls the bitrate extimator with receiver and sender parameters.
+     * @note the bridge is the sender.
+     * @param pkt
+     */
+    @Override
+    public void tccReceived(RTCPTCCPacket pkt)
+    {
+
+        PacketDetail retrievedPacketDetail;
+        for (int receivedSequenceNo: pkt.getPackets().keySet()) {
+            retrievedPacketDetail = outgoingPacketFields
+                    .get(receivedSequenceNo);
+            if (retrievedPacketDetail != null) {
+                /**
+                 * pass SSRC,SendtimeStamp,ReceivedTimeStamp,PacketSize
+                 * to the Remote bitrate Estimator.
+                 *
+                 * @TODO When this is complete, we should have
+                 * a bitrate estimator that uses a different
+                 * delta computation for the sendside of the bridge
+                 */
+
+                long _bitrate = bitrateEstimator
+                        .rtcpTCCRemoteBitrateEstimator((int) pkt.getSourceSSRC(),
+                                retrievedPacketDetail.pktTime,
+                                pkt.getPackets().get(receivedSequenceNo),
+                                retrievedPacketDetail.pktLength);
+
+                /**
+                 * @_bitrate is the remote bandwith of the receiver
+                 * @TODO @_bitrate should be passed to the BandwidthEstimator
+                 */
+
+                /**
+                 * We shouldn't keep packet details once we are done with
+                 * them. Hence, we should delete them
+                 */
+                pkt.getPackets().remove(receivedSequenceNo);
```

Comment:
This might cause problems (deleting an element from the Map while iterating its key set). Perhaps you meant to remove the element from the outgoingPacketFields map?